yarn-project/noir-protocol-circuits/src/crates/private-kernel-lib/src/common.nr

```diff
@@ -325,14 +325,6 @@ pub fn contract_logic(private_call : PrivateCallData, public_inputs : &mut Kerne
          * - Compute the contract_leaf: hash(contract_address, portal_contract_address, function_tree_root)
          * - Hash the contract_leaf with the contract_leaf's sibling_path to get the contract_tree_root
          */
-        
-        // Ensures that if the function is internal, only the contract itself can call it
-        if private_call.call_stack_item.function_data.is_internal {
```

Comment:
Moved it to `validate_call_stack` and only checks for nested calls. The first call can't be internal and is checked in the init kernel.